### PR TITLE
Initial Pescador Benchmark tests

### PR DESCRIPTION
**Pull request by cjacoby**
- Have base function done for without zmq vs with zmq
- Due to pescador [zmq align bug](https://github.com/bmcfee/pescador/issues/22), only doing zmq_stream _with_ copy right now... although it's a manual copy using `np.array`.

### CODE REVIEW

File: testbench_gil.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+import minibench
+
+logging.basicConfig(level=logging.INFO)
+
+
+def theano_test_fx(n_dots):
+    # input vector.
+    x = T.matrix('x')
+    # A fake weight matrix.
+    w = T.matrix('w')
```
Comment (ejhumphrey) [inline testbench_gil.py]: for "true" comparison with how theano gets used, this should probably be a `shared` variable

File: testbench_gil.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+@pytest.fixture
+def npy_sampler_params(benchmark, npys_params):
+    npy_files, params = npys_params
+
+    sampler = minibench.samplers.mux_random_slice(
+        sampler=minibench.samplers.one_npy_random_slice,
+        collec=npy_files,
+        shape=params['slice'],
+        n_samples=100,
```
Comment (ejhumphrey) [inline testbench_gil.py:62]: what's the rationale on `n_samples=100`?

File: testbench_gil.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+@pytest.fixture
+def zmq_sampler_params(benchmark, npys_params):
+    npy_files, params = npys_params
+
+    sampler = minibench.samplers.zmq_random_slice(
+        sampler=minibench.samplers.one_npy_random_slice,
+        collec=npy_files,
+        shape=params['slice'],
+        n_samples=100,
```
Comment (ejhumphrey) [inline testbench_gil.py:77]: ici aussi

File: testbench_gil.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+        lam=params['lam'],
+        working_size=params['working_size'],
+        with_replacement=True)
+    return sampler, params
+
+
+def run_theano_fx(train_fx, sampler, weights):
+    """Make some data and make sure the function runs"""
+    errs = []
+    for sample in sampler:
```
Comment (ejhumphrey) [inline testbench_gil.py:87]: thoughts on sample vs batch (several samples)?

File: testbench_gil.py (hunk 2 above)
Comment (cjacoby) [inline testbench_gil.py:62]: For testing without benchmarking I needed a finite number.

File: testbench_gil.py (hunk 4 above)
Comment (cjacoby) [inline testbench_gil.py:87]: I think either separate tests for batch, or we should just be doing batch... after it's working all the way.

File: testbench_gil.py (hunk 2 above)
Comment (ejhumphrey) [inline testbench_gil.py:62]: +1

File: testbench_gil.py (hunk 4 above)
Comment (ejhumphrey) [inline testbench_gil.py:87]: I think in GIL testing, we want to see that batch generation happens in the background; stress testing is a lil different in that regard.

File: testbench_gil.py (hunk 3 above)
Comment (ejhumphrey) [inline testbench_gil.py:77]: nm given other comment

**Comment by cjacoby** [comment]
@ejhumphrey I fixed the important ones; the rest is work-in-progress.

**Pull request merged**